pkg/controllermanager/controller/backupentry/backup_entry_control.go

```diff
@@ -0,0 +1,468 @@
+		func(backupEntry *gardencorev1alpha1.BackupEntry) (*gardencorev1alpha1.BackupEntry, error) {
+			backupEntry.Status.LastOperation = lastOperation
+			backupEntry.Status.LastError = lastError
+			backupEntry.Status.ObservedGeneration = backupEntry.Generation
+			backupEntry.Status.Seed = seed
+			return backupEntry, nil
+		})
+	if err == nil {
+		o.BackupEntry = newBackupEntry
+	}
+	return err
+}
+
+func (c *defaultControl) removeFinalizer(op *operation.Operation) error {
+	backupEntryFinalizers := sets.NewString(op.BackupEntry.Finalizers...)
+	backupEntryFinalizers.Delete(gardenv1beta1.GardenerName)
+	op.BackupEntry.Finalizers = backupEntryFinalizers.UnsortedList()
+
+	newBackupEntry, err := c.k8sGardenClient.GardenCore().CoreV1alpha1().BackupEntries(op.BackupEntry.Namespace).Update(op.BackupEntry)
+	if err != nil {
+		op.Logger.Errorf("Could not remove finalizer of the BackupEntry: %+v", err.Error())
+		return err
+	}
+	op.BackupEntry = newBackupEntry
+
+	// Wait until the above modifications are reflected in the cache to prevent unwanted reconcile
+	// operations (sometimes the cache is not synced fast enough).
+	return wait.PollImmediate(time.Second, 30*time.Second, func() (bool, error) {
+		backupEntry, err := c.k8sGardenCoreInformers.BackupEntries().Lister().BackupEntries(op.BackupEntry.Namespace).Get(op.BackupEntry.Name)
+		if apierrors.IsNotFound(err) {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		if !sets.NewString(backupEntry.Finalizers...).Has(gardenv1beta1.GardenerName) {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+func computeGracePeriod(config *config.ControllerManagerConfiguration) time.Duration {
+	return time.Hour * 24 * time.Duration(*config.Controllers.BackupEntry.DeletionGracePeriodDays)
+}
+
+func formatError(message string, err error) *gardencorev1alpha1.LastError {
+	return &gardencorev1alpha1.LastError{
+		Description: fmt.Sprintf("%s (%s)", message, err.Error()),
+	}
+}
+
+func nextReconcileScheduleReached(obj *gardencorev1alpha1.BackupEntry, syncPeriod time.Duration) bool {
+	lastOperation := obj.Status.LastOperation
+
+	if lastOperation != nil &&
+		lastOperation.Type == gardencorev1alpha1.LastOperationTypeReconcile &&
+		lastOperation.State == gardencorev1alpha1.LastOperationStateSucceeded {
+
+		earliestNextReconcile := lastOperation.LastUpdateTime.Add(syncPeriod)
+		return time.Now().After(earliestNextReconcile)
+	}
+	return true
+}
+
+func determineSeedForBackupEntry(backupEntry *gardencorev1alpha1.BackupEntry, backupBucket *gardencorev1alpha1.BackupBucket, shootLister gardenv1beta1lister.ShootLister) (string, error) {
+	if backupEntry.Status.Seed != nil && len(*backupEntry.Status.Seed) != 0 {
+		return *backupEntry.Status.Seed, nil
+	}
+
+	// TODO: This is to loadbalance the backupEntry reconciliation across seeds
+	// If shoot persists, Get the seed cluster for shoot controlplane and assign it to backupEntry
```

Comment:
I don't understand these comments. The `BackupEntry` should always exist in the same seed like the `Shoot`, right? What/why do you want to load balance?